async-nats/src/connection.rs

```diff
@@ -30,6 +31,25 @@ pub(crate) trait AsyncReadWrite: AsyncWrite + AsyncRead + Send + Unpin {}
 /// Blanked implementation that applies to both TLS and non-TLS `TcpStream`.
 impl<T> AsyncReadWrite for T where T: AsyncRead + AsyncWrite + Unpin + Send {}
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub enum State {
+    Pending,
+    Connected,
+    Disconnected,
+    Closed,
```

Comment:
Currently, we can't actually reach closed can we?

```suggestion
    Closed,
```